app-next/app/room/[id]/page.jsx

```diff
@@ -0,0 +1,89 @@
+"use client";
+import { useEffect, useState } from "react";
+import { useParams } from "next/navigation";
+import { getRoom, getMessages, sendMessage } from "@/roomActions";
+import Navbar from "@/components/navbar";
+import Footer from "@/components/footer";
+import Chat from "@/components/Chat";
+
+export default function RoomPage() {
+  const params = useParams();
+  const { id } = params;
+
+  const [room, setRoom] = useState(null);
+  const [messages, setMessages] = useState([]);
+  const [videoUrl, setVideoUrl] = useState("");
+
+  useEffect(() => {
+    const fetchRoomData = async () => {
+      try {
+        const roomData = await getRoom(id);
+        if (roomData) {
+          setRoom(roomData);
+          setVideoUrl(roomData.video_url);
+        }
+        const messagesData = await getMessages(id);
+        setMessages(messagesData || []);
+      } catch (error) {
+        console.error("Error fetching room data:", error);
+      }
+    };
+
+    fetchRoomData();
+  }, [id]);
```

Comment:
@l0rians It would be simpler here if you convert this page into a server component. Remove this `useEffect` and, instead just fetch here directly, and remove `"use client"` from top.